### ISSUE TITLE
Add support for BFSTM types of files

### ISSUE THREAD

**Issue by MVDW-Java** [kenrick95/nikku#9]
Hello Kenrick95, 

I would like to suggest to add support for BFSTM.
This file is used in most Nintendo Switch and Wii U Games.

Thank you for the amazing work you have done making a web-based media player!

Kind regards,
MVDW.

**Comment by kenrick95** [comment]
You're welcome!

Unfortunately, I'm not currently able to support this idea and I'll left this issue open for now. Thanks

**Comment by Extrasklep** [comment]
I might be able to try doing this after writing a BFSTM decoder in my own C++ BRSTM tools to have a good understanding of it but the JS module would have to support different header reading functions and file type detection

**Comment by kenrick95** [comment]
@Extrasklep Nice, thanks!

I would say that for a simple version we could just check using the file extension.

https://github.com/kenrick95/nikku/blob/773ab8a4b60c45223c73dc9d9dcaf4f68c6d10b4/index.html#L26

Right now in the index.html, it only accepts ".brstm". So we need to extend this too.

For implementation, I suggest that this bfstm decoder be a separate "package" in the `src/` folder so that we don't make the brstm decoder be more complex than what it already is. What do you think?

**Comment by Extrasklep** [comment]
Yeah I guess it should be a separate file but it would also be stupid to completely separate it from the current package and copy the same audio decoding code with it when it could just use the same functions
In my BRSTM tools I just split each format's reading function into a separate file and the main read function does file type detection, calls the appropriate read function, and the program using it will just use the same audio decoding functions but I'm guessing this JS class stuff works a bit different than that, I'm not really into classes and object oriented programming

**Comment by kenrick95** [comment]
I see, yeah I think it's your call then on whether to make it a separate folder or not. I suggested that cause I would think that almost nothing is sharable across these two modules. If they do share lots of logic, then making it in a same folder is a good idea.

Sorry for the OOP, my original goal of this project was to make it work without regarding what paradigm is the best, I just used one that I like 😅 

Thanks :)

**Comment by Extrasklep** [comment]
The class constructor could do file type detection and then other format specific things could be moved into a separate format reading function or am I wrong?

**Comment by kenrick95** [comment]
Yup looks good :)
